src/pages/questions/useQuestionBuffer.ts

```diff
@@ -87,21 +143,58 @@ function reducer(state, action) {
         answers: [
           ...state.answers,
           {
-            insight_id: answeredQuestion?.insight_id,
-            barcode: answeredQuestion?.barcode,
-            insight_type: answeredQuestion?.insight_type,
-            value: answeredQuestion?.value,
+            ...answeredQuestion,
             validationValue: action.payload.value,
+            isPending: true,
+            sendingTime: new Date().getTime() + action.payload.pendingDelay,
           },
         ],
         remainingQuestionNb: state.remainingQuestionNb - 1,
         // skipped ids is used to correctly compute remainingQuestionNb after each data fetching
         skippedIds: [
           ...state.skippedIds,
-          ...(action.payload.value === -1 ? [action.payload.insight_id] : []),
+          ...(action.payload.value === -1 ? [action.payload.insightId] : []),
         ],
       };
 
+    case "sendAnswers":
+      const minDate = new Date().getTime();
+
+      const newAnswers = state.answers.map((answer) => {
```

Comment:
what about filtering (state.answers.filter(..)) the ones that need to be annotated (isPending && sendingTime <= minDate), and use a Promise.all(filteredArray => robotoff.annotate()).then(()=> return {}).catch(()=> return state), something like that...  calling the annotate inside the map makes me feel like doing two different things at the same time